plugins/arOidcPlugin/lib/oidcUser.class.php

```diff
@@ -262,13 +275,135 @@ public function logout()
             // so $this->oidcClient->signOut will fail.
             // https://github.com/dexidp/dex/issues/1697
             try {
-                $this->oidcClient->signOut($idToken, $logoutRedirectUrl);
+                // Get saved session provider id.
+                $providerId = $this->getSessionProviderId();
+                // Set provider details in the OIDC client using provider id.
+                if (true === $result = $this->setOidcProviderDetails($providerId)) {
+                    $this->oidcClient->signOut($idToken, $logoutRedirectUrl);
+                }
+                $this->setProviderId();
             } catch (Exception $e) {
+                $this->setProviderId();
```

Comment:
Is this repeated function call inside the catch a typo?

Reply:
I was trying to ensure that setProviderId would be run even if there was an exception generated in signOut().

I am wondering now if I could just move getSessionProviderID() and setProviderId() to before the try{}. I will have a closer look.

Reply:
Updated and tested. See latest commit.